runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngine.kt

```diff
@@ -70,20 +70,29 @@ private fun OkHttpEngineConfig.buildClient(): OkHttpClient {
         followSslRedirects(false)
 
         // see: https://github.com/ktorio/ktor/issues/1708#issuecomment-609988128
+        // TODO disable this once better transient exception handling is in place
         retryOnConnectionFailure(true)
 
         connectTimeout(config.connectTimeout.toJavaDuration())
         readTimeout(config.socketReadTimeout.toJavaDuration())
         writeTimeout(config.socketWriteTimeout.toJavaDuration())
 
-        // use our own pool configured with the settings taken from config
+        // use our own pool configured with the timeout settings taken from config
         val pool = ConnectionPool(
-            maxIdleConnections = config.maxConnections.toInt(),
+            maxIdleConnections = 5, // The default from the no-arg ConnectionPool() constructor
```

Comment:
Question: Should this be a knob? I don't see any documentation on how this parameter is used internally to okhttp.

Reply:
From what I've observed, this is the maximum number of connections that will be left idle in the pool. When another connection would be returned to the pool and that would exceed the `maxIdleConnections`, the oldest connection is found and evicted.

Exposing a knob here might help with sporadic traffic. Keeping a large reserve of idle connections may accelerate rapid bursts of activity after periods of inactivity. That said, it's hyper-specific. I vote for not exposing this knob for now and revisiting if we get feedback about it.